create_builder_test.go

```diff
@@ -160,6 +168,31 @@ func testCreateBuilder(t *testing.T, when spec.G, it spec.S) {
 				err := subject.CreateBuilder(context.TODO(), opts)
 				h.AssertError(t, err, "buildpack from URI 'https://example.fake/bp-one.tgz' has version '1.2.3' which does not match version '0.0.0' from builder config")
 			})
+
+			it("should fail when buildpack is not compatible with lifecycle", func() {
```

Comment:
Same comment as for `build_test.go`, as long as we have one test where we error if saving the builder errors, I don't think we need to retest all of the validations from `builder_test.go`.

Reply:
We couldn't find the duplicate test. Could you clarify? There are other errors tested in builder_test.go but not this one specifically.

Reply:
Did you mean `build_test.go`?

Reply:
I guess it isn't currently in `builder_test.go` but it seems analogous to https://github.com/buildpack/pack/blob/feature/254-validate-buildpack-api/builder/builder_test.go#L437 to me. This is where I would expect to look for this level of coverage.

Reply:
Makes sense. Moved test.